Assets/MixedRealityToolkit.Services/InputAnimation/InputAnimation.cs

```diff
@@ -0,0 +1,710 @@
+        {
+            private InputAnimation animation;
+
+            public AnimationCurveEnumerable(InputAnimation animation)
+            {
+                this.animation = animation;
+            }
+
+            public IEnumerator<AnimationCurve> GetEnumerator()
+            {
+                yield return animation.handTrackedCurveLeft;
+                yield return animation.handTrackedCurveRight;
+                yield return animation.handPinchCurveLeft;
+                yield return animation.handPinchCurveRight;
+                foreach (var curves in animation.handJointCurvesLeft.Values)
+                {
+                    yield return curves.PositionX;
+                    yield return curves.PositionY;
+                    yield return curves.PositionZ;
+                    yield return curves.RotationX;
+                    yield return curves.RotationY;
+                    yield return curves.RotationZ;
+                    yield return curves.RotationW;
+                }
+                foreach (var curves in animation.handJointCurvesRight.Values)
+                {
+                    yield return curves.PositionX;
+                    yield return curves.PositionY;
+                    yield return curves.PositionZ;
+                    yield return curves.RotationX;
+                    yield return curves.RotationY;
+                    yield return curves.RotationZ;
+                    yield return curves.RotationW;
+                }
+                yield return animation.cameraCurves.PositionX;
+                yield return animation.cameraCurves.PositionY;
+                yield return animation.cameraCurves.PositionZ;
+                yield return animation.cameraCurves.RotationX;
+                yield return animation.cameraCurves.RotationY;
+                yield return animation.cameraCurves.RotationZ;
+                yield return animation.cameraCurves.RotationW;
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return this.GetEnumerator();
+            }
+        }
+
+        private IEnumerable<AnimationCurve> AnimationCurves => new AnimationCurveEnumerable(this);
+
+        /// <summary>
+        /// Find an index i in the sorted events list, such that events[i].time <= time < events[i+1].time.
+        /// </summary>
+        /// <returns>
+        /// 0 <= i < eventCount if a full interval could be found.
+        /// -1 if time is less than the first event time.
+        /// eventCount-1 if time is greater than the last event time.
+        /// </returns>
+        /// <remarks>
+        /// Uses binary search.
+        /// </remarks>
+        private static int FindKeyframeInterval(AnimationCurve curve, float time)
+        {
+            var keys = curve.keys;
+            int lowIdx = -1;
+            int highIdx = keys.Length;
+            while (lowIdx < highIdx - 1)
+            {
+                int midIdx = (lowIdx + highIdx) >> 1;
```

Comment:
Is there a measureable difference in the perf of a shift vs simple divide by 2?

Reply:
That's my C coder instinct shining through 😉 
Probably doesn't matter, but doesn't hurt either.